docs/create-your-own-modern-controllers.md

```diff
@@ -35,7 +35,37 @@ Note that you must return a `Response` object, but this can be a `JsonResponse`
 
 > This controller is the same than the ones used in Back Office. 
 
-Now we have created your controller, you need to declare a route. A route map an action of your controller to an URI.
+You must enable the autoloading for this Controller. For example using a `composer.json` file for your module.
+
+*Example using PSR-4 namespacing:*
+
+- use namespace for your Controller file
+
+```php
+// modules/your-module/controller/DemoController.php
+
+namespace MyModule\Controller;
+
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+```
+
+- configure composer to autoload this namespace
+
+```
+{
+  "name": "you/your-module",
+  "description": "...",
+  "autoload": {
+    "psr-4": {
+      "MyModule\\Controller\\": "controller/"
```

Comment:
i would suggest putting modern controller under `src/Controller` directory in your module.

Reply:
Pushing a convention can help me later to provide auto wiring, but I'm not sure we should do it :/

Reply:
If we move it into `src`, we might confuse module developers because the current doc asks them to put controllers into `controllers` : http://doc.prestashop.com/display/PS16/Creating+a+first+module#Creatingafirstmodule-FilestructureforaPrestaShopmodule

But we can also upgrade the doc :) . Such Symfony Controllers will only work for PrestaShop > 1.7.5 anyway